app/src/main/java/sistemas/puc/com/finantialapp/FetchIndicesTask.java

```diff
@@ -0,0 +1,212 @@
+    private static final String IPCA_URL = "http://servicodados.ibge.gov.br/api/v2/conjunturais/1419/periodos/-1/indicadores";
+    private static final String INPC_URL = "http://servicodados.ibge.gov.br/api/v2/conjunturais/1100/periodos/-1/indicadores";
+
+    private final String LOG_TAG = FetchIndicesTask.class.getSimpleName();
+    private Context m_context;
+
+    public FetchIndicesTask(Context context) {
+        m_context = context;
+    }
+
+    @Override
+    protected void onPreExecute(){}
+
+    @Override
+    protected List<ContentValues> doInBackground(Void... voids) {
+
+        String selicJsonStr = getJsonStr(SELIC_URL);
+        String inpcJsonStr = getJsonStr(INPC_URL);
+        String ipcaJsonStr = getJsonStr(IPCA_URL);
+
+        //placeholders
+
+        ContentValues CDI = new ContentValues();
+        setupIndiceCV(CDI, IndiceEnum.CDI);
+        CDI.put(FinantialContract.IndiceEntry.COLUMN_INDICE_MONTH_RATE, 0.0);
+        CDI.put(FinantialContract.IndiceEntry.COLUMN_INDICE_YEAR_RATE, 0.0);
+        CDI.put(FinantialContract.IndiceEntry.COLUMN_INDICE_DATE, "201709");
+
+        ContentValues poupanca = new ContentValues();
+        setupIndiceCV(poupanca, IndiceEnum.POUPANCA);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_MONTH_RATE, 0.0);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_YEAR_RATE, 0.0);
+        poupanca.put(FinantialContract.IndiceEntry.COLUMN_INDICE_DATE, "201709");
+
+        ContentValues IGPM = new ContentValues();
+        setupIndiceCV(IGPM, IndiceEnum.IGPM);
+        IGPM.put(FinantialContract.IndiceEntry.COLUMN_INDICE_MONTH_RATE, 0.0);
+        IGPM.put(FinantialContract.IndiceEntry.COLUMN_INDICE_YEAR_RATE, 0.0);
+        IGPM.put(FinantialContract.IndiceEntry.COLUMN_INDICE_DATE, "201709");
+
+        try {
+            List<ContentValues> result = new ArrayList<>();
+            result.add(getSelicFromJson(selicJsonStr));
+            result.add(CDI);
+            result.add(getIndiceFromIBGEJson(ipcaJsonStr, IndiceEnum.IPCA));
+            result.add(getIndiceFromIBGEJson(inpcJsonStr, IndiceEnum.INPC));
+            result.add(IGPM);
+            result.add(poupanca);
+            return result;
+        } catch (JSONException e) {
+            Log.e(LOG_TAG, e.getMessage(), e);
+            e.printStackTrace();
+        }
+
+        return null;
+    }
+
+    @Override
+    protected void onProgressUpdate(Void... voids) {}
+
+    @Override
+    protected void onPostExecute(List<ContentValues> result) {
+        if (result.size() > 0) {
+            m_context.getContentResolver().delete
+                    (FinantialContract.IndiceEntry.CONTENT_URI, null, null);
+            ContentValues[] cvArray = result.toArray(new ContentValues[result.size()]);
+            m_context.getContentResolver().bulkInsert
+                    (FinantialContract.IndiceEntry.CONTENT_URI, cvArray);
+        }
+    }
+
+    private String getJsonStr(String URL) {
```

Comment:
Dont you think that this function could be extracted? It is the same for all json requests